app/models/manager_refresh/inventory_collection_default.rb

```diff
@@ -1,4 +1,29 @@
 class ManagerRefresh::InventoryCollectionDefault
+  VM_OR_TEMPLATE_RECONNECT_BLOCK = lambda do |inventory_collection, inventory_objects_index, attributes_index|
+    relation = inventory_collection.model_class.where(:ems_id => nil)
+
+    return if relation.count <= 0
+
+    inventory_objects_index.each_slice(100) do |batch|
+      relation.where(inventory_collection.manager_ref.first => batch.map(&:first)).each do |record|
```

Comment:
@agrare so we are doing reconnect using :ems_ref, which is ok for clouds and also RHEV (right @borod108 ?)

So I would just keep it this way and when we play with VMware, we would figure out what to do. E.g. we can just use uid_ems as unique attribute? It's unique for VMware, so we can use it both for matching and reconnect?

Reply:
@Ladas according to our "API" (aka general knowledge haha)`:ems_ref` isn't unique across ems_ids, even if it happens to be for cloud providers.  Most just set `ems_ref = uid_ems = some unique id` but RHEV actually has two different values for ems_ref and uid_ems.  Why not just use uid_ems as the manager_ref and be done with it?

Reply:
@agrare right, that is what I was thinking, but I would rather keep it this way for backport (switching manager_ref globally could break some specs and we might need to do minor parser changes)

and for upstream, we could just switch to :manager_ref => [:uid_ems], for both Vm and MiqTemplate, does that sound good?

Reply:
👍 works for me

Reply:
If I got this right, you're talking about some change, that's not fully related to the current PR/issue?

Reply:
@AlexanderZagaynov yeah, we can ignore it for now. We will do changes on other PR, to make sure this works for VMware